drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/constraints/RulebookOperator.java

```diff
@@ -19,17 +21,42 @@ default ConstraintOperator asConstraintOperator() {
         return this;
     }
 
-    RulebookOperator EQUAL = new OperatorWrapper(Index.ConstraintType.EQUAL);
-    RulebookOperator NOT_EQUAL = new OperatorWrapper(Index.ConstraintType.NOT_EQUAL);
-    RulebookOperator GREATER_THAN = new OperatorWrapper(Index.ConstraintType.GREATER_THAN);
-    RulebookOperator GREATER_OR_EQUAL = new OperatorWrapper(Index.ConstraintType.GREATER_OR_EQUAL);
-    RulebookOperator LESS_THAN = new OperatorWrapper(Index.ConstraintType.LESS_THAN);
-    RulebookOperator LESS_OR_EQUAL = new OperatorWrapper(Index.ConstraintType.LESS_OR_EQUAL);
+    static RulebookOperator newEqual() {
+        return new OperatorWrapper(new RulebookConstraintOperator(Index.ConstraintType.EQUAL));
+    }
+
+    static RulebookOperator newNotEqual() {
+        return new OperatorWrapper(new RulebookConstraintOperator(Index.ConstraintType.NOT_EQUAL));
+    }
+
+    static RulebookOperator newGreaterThan() {
+        return new OperatorWrapper(new RulebookConstraintOperator(Index.ConstraintType.GREATER_THAN));
+    }
+
+    static RulebookOperator newGreaterOrEqual() {
+        return new OperatorWrapper(new RulebookConstraintOperator(Index.ConstraintType.GREATER_OR_EQUAL));
+    }
+
+    static RulebookOperator newLessThan() {
+        return new OperatorWrapper(new RulebookConstraintOperator(Index.ConstraintType.LESS_THAN));
+    }
+
+    static RulebookOperator newLessOrEqual() {
+        return new OperatorWrapper(new RulebookConstraintOperator(Index.ConstraintType.LESS_OR_EQUAL));
+    }
+
+    default void setConditionContext(RuleGenerationContext ruleContext, Map<?, ?> expression) {
+        if (this instanceof OperatorWrapper operatorWrapper) {
+            operatorWrapper.setConditionContext(ruleContext, expression);
+        } else {
+            // do nothing
+        }
+    }
 
     class OperatorWrapper implements RulebookOperator {
```

Comment:
This wrapper seems no longer necessary, couldn't `RulebookConstraintOperator` directly implement `RulebookOperator`?

Reply:
Thanks, done.